File: 백준/Bronze/15829. Hashing/Hashing.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const L = Number(input[0]);
const S = input[1];

const r = 31n;
const M = 1234567891n;

let totalHash = 0n;
let power = 1n;

for (let i = 0; i < L; i++) {
    const charVal = BigInt(S.charCodeAt(i) - 96);
    totalHash = (totalHash + charVal * power) % M;
    power = (power * r) % M;
}

console.log(totalHash.toString());

File: 백준/Bronze/15829. Hashing/Hashing.py
import sys
input = sys.stdin.readline

L = int(input())
S = input().strip()

r = 31
M = 1234567891

total_hash = 0

for i in range(L):
    char_val = ord(S[i]) - ord('a') + 1
    total_hash += char_val * (r ** i)

print(total_hash % M)